File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.js
// Last updated: 4/29/2026, 10:30:53 AM
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function(nums1, nums2) {
    const map = {};
    for(const num1 of nums1) {
        if(map[num1] === undefined) {
            map[num1] = true;
        }
    }
    console.log(map);
    const set = new Set();
    for(const num2 of nums2) {
        if(map[num2]) {
            set.add(num2);
        }
    }
    return [...set];
};

File: 349-IntersectionOfTwoArrays/349-IntersectionOfTwoArrays.py
# Last updated: 4/29/2026, 10:30:58 AM
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        from collections import defaultdict
        dict = defaultdict(lambda: None)
        for num1 in nums1:
            if dict[num1] == None:
                dict[num1] = 1
            else: 
                dict[num1] += 1
        print(dict)
        res_set = set()
        for num2 in nums2:
            val = dict[num2]
            if val != None and val >= 0:
                res_set.add(num2)
                dict[num2] -= 1
        return list(res_set)